pkg/client/dtclient/client_settings.go

```diff
@@ -187,6 +187,12 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 		}
 	}
 
+	if objectID, err := d.findObjectWithMatchingConstraints(ctx, obj); err != nil {
+		return DynatraceEntity{}, err
+	} else if objectID != nil {
+		obj.OriginObjectId = objectID.ObjectId
```

Comment:
having debug logging for this case would be helpful I think

```suggestion
		log.Debug("Updating existing object %q with matching unique key properties", objectID.ObjectId")     
		obj.OriginObjectId = objectID.ObjectId
```